### DISCUSSION TITLE
[Feature] Seperate port or IP for link sharing

### DISCUSSION THREAD

**Discussion by NicholasFlamy** [immich-app/immich#4785] [Feature Request]
### The feature

I'd like my links to use a different port that way I don't have to expose my instance to the internet and can use a VPN (I currently have Wireguard set up so that I can access my server from away) to login but my share links can be on a port open to the internet without exposing my immich instance. I don't want public access to all of immich because of the security risks of this (if the login page is accessible it can be hacked, if it isn't accessible then there is almost no way to hack it.)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
You can already do this by just white listing the required routes for sharing to work on your reverse proxy and having the proxy utilize a different port.

**Reply by NicholasFlamy** [reply]
That's exactly what I've been researching, it's just that I'm not that good at it yet.

**Reply by jrasm91** [reply]
Ah, I see. I think you can make a second entry in your proxy config also map back to immich. And then you just need to figure out how to block non authorized routes. That will depend on what reverse proxy you are using.

**Reply by NicholasFlamy** [reply]
So I got Nginx Proxy Manager going. Now I need to figure out how to only allow a specific directory/location of the webpage. I can see the custom location configuration, just gotta figure out how to deny all except /share

**Reply by jrasm91** [reply]
FYI - might be worth asking on discord as I think other people have done something similar.

**Reply by NicholasFlamy** [reply]
I posted the question on Discord. Now to wait.

**Comment by NicholasFlamy** [comment]
#4852 is what I need to get this working.

(Commenting so that anyone who comes along will see the useful post.)
